File: Data/States/PlanState.cs
namespace Data.States
{
    public class PlanState
    {
        public const int Inactive = 1;
        public const int Executing = 2;
        public const int Deleted = 3;
        public const int Active = 4;
        
        public static int StringToInt(string str)
        {
            int planState = 1;

            switch (str)
            {
                case "Inactive":
                    planState = Inactive;
                    break;

                case "Executing":
                    planState = Executing;
                    break;

                case "Active":
                    planState = Active;
                    break;

                case "Deleted":
                    planState = Deleted;
                    break;
            }

            return planState;
        }

        public static string IntToString(int planState)
        {
            string state = "Inactive";

            switch (planState)
            {
                case Inactive:
                    state = "Inactive";
                    break;

                case Active:
                    state = "Active";
                    break;

                case Deleted:
                    state = "Deleted";
                    break;

                case Executing:
                    state = "Executing";
                    break;
            }

            return state;
        }
    }
}

File: Hub/HubWeb/Scripts/app/filters/PlanState.ts
/// <reference path="../_all.ts" />

/*
    The filter converts numeric value to state name
*/
module dockyard {
    'use strict';
    app.filter('PlanState', () =>
        (input : string): string => {
            switch (input)
            {
                case model.PlanState.Executing:
                    return "Executing";
                case model.PlanState.Inactive:
                    return "Inactive";
                case model.PlanState.Active:
                    return "Active";
                default:
                    return "Inactive";
            }
        });
}